src/ya-comiste-rust/server/src/commerce/dal/products.rs

```diff
@@ -0,0 +1,297 @@
+use crate::arangodb::ConnectionPool;
+use crate::commerce::model::errors::ModelError;
+use crate::commerce::model::products::{
+    ClientLocale, PriceSortDirection, Product, ProductMultilingualInput,
+};
+use serde_json::json;
+
+/// Takes care of creating the product inside ArangoDB.
+pub(in crate::commerce) async fn create_product(
+    pool: &ConnectionPool,
+    product_multilingual_input: &ProductMultilingualInput,
+) -> Result<Product, ModelError> {
+    let db = pool.db().await;
+
+    // TODO: categories
+    // TODO: dynamic `unit_label`
+
+    let en_us = product_multilingual_input.en_us.as_ref();
+    let es_mx = product_multilingual_input.es_mx.as_ref();
+
+    let insert_aql = arangors::AqlQuery::builder()
+        .query(
+            r#"
+            LET translations = (
+              FOR t IN @translations
+                RETURN {
+                  [t.lang]: {
+                    name: t.name,
+                    description: t.description
+                  }
+                }
+            )
+
+            INSERT {
+              images: @product_images,
+              unit_label: "product_units/piece",
+              active: false,
+              created: DATE_ISO8601(DATE_NOW()),
+              updated: DATE_ISO8601(DATE_NOW()),
+              price: {
+                currency: "mxn",
+                unit_amount: @product_price_unit_amount
+              },
+              translations: MERGE(translations)
+            } INTO products
+            LET product = NEW
+            RETURN UNSET(
+              MERGE_RECURSIVE(product, product.translations[@product_language]),
+              "translations"
+            )
+            "#,
+        )
+        .bind_var("product_language", String::from("en_US")) // TODO
```

Comment:
🤔 This is good for fetching on FE but what if BO needs to fetch both language versions at the same time? (for product editation for example)

Reply:
I should probably add a new field `translations` to the output because it makes sense to return the title in the locale we have BO/FE but also return multiple translation versions if needed (for the edits for example).